code/game/turfs/open/space/space.dm

```diff
@@ -31,49 +31,36 @@
 /**
  * Space Initialize
  *
- * Doesn't call parent, see [/atom/proc/Initialize]
+ * Doesn't call parent, see [/atom/proc/Initialize].
+ * When adding new stuff to /atom/Initialize, /turf/Initialize, etc
+ * don't just add it here unless space actually needs it.
  */
 /turf/open/space/Initialize(mapload)
 	SHOULD_CALL_PARENT(FALSE)
-	icon_state = SPACE_ICON_STATE
+	icon_state = SPACE_ICON_STATE(x, y, z)
 	air = space_gas
-	vis_contents.Cut() //removes inherited overlays
-	visibilityChanged()
 
 	if(flags_1 & INITIALIZED_1)
 		stack_trace("Warning: [src]([type]) initialized multiple times!")
 	flags_1 |= INITIALIZED_1
 
-	if (length(smoothing_groups))
-		sortTim(smoothing_groups) //In case it's not properly ordered, let's avoid duplicate entries with the same values.
-		SET_BITFLAG_LIST(smoothing_groups)
-	if (length(canSmoothWith))
-		sortTim(canSmoothWith)
-		if(canSmoothWith[length(canSmoothWith)] > MAX_S_TURF) //If the last element is higher than the maximum turf-only value, then it must scan turf contents for smoothing targets.
-			smoothing_flags |= SMOOTH_OBJ
-		SET_BITFLAG_LIST(canSmoothWith)
-
 	var/area/our_area = loc
 	if(our_area.area_has_base_lighting && always_lit) //Only provide your own lighting if the area doesn't for you
-		add_overlay(GLOB.fullbright_overlay)
-
-	if(requires_activation)
-		SSair.add_to_active(src, TRUE)
-
-	if (light_system == STATIC_LIGHT && light_power && light_range)
-		update_light()
-
-	if (opacity)
-		directional_opacity = ALL_CARDINALS
-
-	var/turf/T = SSmapping.get_turf_above(src)
```

Comment:
Moved to mapload, couldn't see any problems from it. Where this is important is openspace, which already adds `/datum/element/turf_z_transparency`, which performs this same linking.

Reply:
are you sure about that? what if openspace inits, something spawns us in during mapload, and we try and link? wouldn't this fail?

Reply:
Can you explain the actual scenario that'll come up in game and what I should expect to see? I can test it but these were like 6 seconds of total init time

Reply:
Nothing immediately comes to mind but it's something I could see someone accidentally doing.
To be clear I don't think the optimization is improper, just that you should be careful to mark out the fragile bits of this removal so future people don't end up in hell